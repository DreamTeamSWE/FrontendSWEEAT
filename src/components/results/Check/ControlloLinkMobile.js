import React from 'react';
import './../css/cards.css';

export default function MobileLink(props){
  if(props.sitoweb !== null){
  return(
    <div className="links">
      <a className='website' href={props.sitoweb} target="_blank"><span>Vai al sito</span></a>
      <a className='location' href={'https://www.google.com/maps/?q=' + props.latitudine + "," + props.longitudine} target="_blank"><span>Apri su GMaps</span></a>
    </div>
  )
  }
  else{
    return(
      <a className={'location_only'} href={'https://www.google.com/maps/?q=' + props.latitudine + "," + props.longitudine} target="_blank"><span>Apri su GMaps</span></a>
    )
  }
}
