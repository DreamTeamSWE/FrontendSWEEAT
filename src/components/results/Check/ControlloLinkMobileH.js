import React from 'react';

export default function MobileLink(props){
  if(props.sitoweb !== null){
  return(
    <div className="linksHome">
      <a className='websiteHome' href={props.sitoweb} target="_blank"><span>Vai al sito</span></a>
      <a className='locationHome' href={'https://www.google.com/maps/?q=' + props.latitudine + "," + props.longitudine} target="_blank"><span>Apri su GMaps</span></a>
    </div>
  )
  }
  else{
    return(
      <a className={'location_onlyHome'} href={'https://www.google.com/maps/?q=' + props.latitudine + "," + props.longitudine} target="_blank"><span>Apri su GMaps</span></a>
    )
  }
}
