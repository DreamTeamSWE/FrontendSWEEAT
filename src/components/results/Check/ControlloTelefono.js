import React from 'react';
import telefono from '../../../images/phone-call.png';
import './../css/cards.css';

export default function CheckPhone(props) {
  if (props.phone !== "") {
    return (
      <div className="telefono">
        <img src={telefono} alt="Chiama" />
        <a href={"tel:" + props.phone}>{props.phone}</a>
      </div>
    )
  }
  return (
    <p />
  );
}
