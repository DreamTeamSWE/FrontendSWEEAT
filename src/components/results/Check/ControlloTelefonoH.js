import React from "react";
import telefono from "../../../images/phone-call.png";

export default function CheckPhone(props) {
  if (props.phone !== "") {
    return (
      <div className="telefonoHome">
        <img src={telefono} alt="Chiama" />
        <a href={'tel:' + props.phone}>{props.phone}</a>
      </div>
    );
  }
  return <p className="hidden"></p>;
}
