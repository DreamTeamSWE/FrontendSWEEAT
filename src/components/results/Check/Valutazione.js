import React from "react";

export default function Valutazione(props) {
  if (4.5 < props.val && props.val <= 5) {
    return <p>Eccellente</p>;
  }
  if (4 < props.val && props.val <= 4.5) {
    return <p>Ottimo</p>;
  }
  if (3.5 < props.val && props.val <= 4) {
    return <p>Buono</p>;
  }
  if (3 < props.val && props.val <= 3.5) {
    return <p>Punteggio medio</p>;
  }
  if (2.5 < props.val && props.val <= 3) {
    return <p>Discreto</p>;
  }
  if (2 < props.val && props.val <= 2.5) {
    return <p>Sufficiente</p>;
  }
  if (1.5 < props.val && props.val <= 2) {
    return <p>Scarso</p>;
  }
  if (1 < props.val && props.val <= 1.5) {
    return <p>Pessimo</p>;
  }
  return <p>Indefinibile</p>;
}
