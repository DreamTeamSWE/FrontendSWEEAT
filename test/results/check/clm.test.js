import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import ControlloLinkMobile from '../../../src/components/results/Check/ControlloLinkMobile';
import ControlloLinkMobileH from '../../../src/components/results/Check/ControlloLinkMobileH';
import ControlloTelefono from '../../../src/components/results/Check/ControlloTelefono';
import ControlloTelefonoH from '../../../src/components/results/Check/ControlloTelefonoH';
import Valutazione from '../../../src/components/results/Check/Valutazione';
import {render, screen} from '@testing-library/react';
import {BrowserRouter as Router} from 'react-router-dom';
import '@testing-library/jest-dom';

configure({adapter: new Adapter()});

/*describe('render cards', () => {
  render(
  <Router>
  <Cards/>
  const card = shallow(<Cards/>);
  expect(card.find(Link)).toBe('');
  expect(card).toMatchSnapshot();
  </Router>
  )
});*/


describe('Check render controllo link mobile', () => {
  beforeEach(() =>{
  render(
      <ControlloLinkMobile />
  )});

  test('Check render results', () =>{
    const clm = shallow(<ControlloLinkMobile/>);
    expect(clm).toMatchSnapshot();;
  })

});

describe('Check render controllo link mobile', () => {
  beforeEach(() =>{
  render(
      <ControlloLinkMobileH />
  )});

  test('Check render results', () =>{
    const clmh = shallow(<ControlloLinkMobileH/>);
    expect(clmh).toMatchSnapshot();;
  })

});

describe('Check render controllo link mobile', () => {
  beforeEach(() =>{
  render(
      <ControlloTelefono />
  )});

  test('Check render results', () =>{
    const ct = shallow(<ControlloTelefono/>);
    expect(ct).toMatchSnapshot();;
  })

});

describe('Check render controllo link mobile', () => {
  beforeEach(() =>{
  render(
      <ControlloTelefonoH />
  )});

  test('Check render results', () =>{
    const cth = shallow(<ControlloTelefonoH/>);
    expect(cth).toMatchSnapshot();;
  })

});

describe('Check render controllo link mobile', () => {
  beforeEach(() =>{
  render(
      <Valutazione />
  )});

  test('Check render results', () =>{
    const val = shallow(<Valutazione/>);
    expect(val).toMatchSnapshot();;
  })

});
