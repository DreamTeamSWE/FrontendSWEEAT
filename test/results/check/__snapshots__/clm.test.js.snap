// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check render controllo link mobile Check render results 1`] = `ShallowWrapper {}`;

exports[`Check render controllo link mobile Check render results 2`] = `ShallowWrapper {}`;

exports[`Check render controllo link mobile Check render results 3`] = `ShallowWrapper {}`;

exports[`Check render controllo link mobile Check render results 4`] = `ShallowWrapper {}`;

exports[`Check render controllo link mobile Check render results 5`] = `ShallowWrapper {}`;
